types/v1/util_test.go

```diff
@@ -0,0 +1,82 @@
+package v1
+
+import (
+	"errors"
+	"reflect"
+	"testing"
+)
+
+func TestParseAndSubstract(t *testing.T) {
+	cases := map[string]struct {
```

Comment:
I am finding it difficult to understand the test scenarios.
Will it be good to elaborate via the test case name, better variable names if possible.